### PR TITLE
Should fix #234 + refactor domain to hostname

### LINKED ISSUE #234: [AP] Search a user on mastodon gives 503 error
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-22T10:03:28Z" title="Friday, March 22nd 2019, 11:03:28 am +01:00">Mar 22, 2019</time>_
_Closed <time datetime="2019-03-25T16:05:59Z" title="Monday, March 25th 2019, 5:05:59 pm +01:00">Mar 25, 2019</time>_
---

**Describe the bug**
We have the backend deployed which now should be able to return results for webfinger searches.

E.g. if I type in `https://nitro-staging.human-connection.org/api/.well-known/webfinger?resource=acct:bob-der-baumeister` into the URL I get:
```json
// 20190322110313
// https://nitro-staging.human-connection.org/api/.well-known/webfinger?resource=acct:bob-der-baumeister

{
  "subject": "acct:bob-der-baumeister@nitro-backend.human-connection",
  "links": [
    {
      "rel": "self",
      "type": "application/activity+json",
      "href": "https://nitro-backend.human-connection/users/bob-der-baumeister"
    }
  ]
}
```

However, if I search for this seeded user through Mastodon, I get a 503 HTTP error.

**To Reproduce**
1. Visit a mastodon instance
2. Click on "Search"
3. Type in `bob-der-baumeister@nitro-staging.human-connection.org/api/`
4. See 503 HTTP error

**Expected behavior**
I would expect to see the user "Bob der Baumeister" in the result set.

**Screenshots**
2019-03-22-110425_1920x1080_scrot

### PR DESCRIPTION
**Pull request by Tirokk**
> [Mastercuber](https://github.com/Mastercuber) **Authored by [Mastercuber](https://github.com/Mastercuber)**
_<time datetime="2019-03-25T12:18:06Z" title="Monday, March 25th 2019, 1:18:06 pm +01:00">Mar 25, 2019</time>_
_Merged <time datetime="2019-03-25T16:05:59Z" title="Monday, March 25th 2019, 5:05:59 pm +01:00">Mar 25, 2019</time>_
---

close #234

### CODE REVIEW

File: backend/src/activitypub/ActivityPub.js [hunk 1]
```diff
@@ -33,7 +32,7 @@ export default class ActivityPub {
     if (!activityPub) {
       dotenv.config()
       const url = new URL(process.env.GRAPHQL_URI)
-      activityPub = new ActivityPub(url.hostname || 'localhost', url.port || 4000, url.origin)
```
Comment (Tirokk) [inline backend/src/activitypub/ActivityPub.js:36]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-25T13:47:15Z" title="Monday, March 25th 2019, 2:47:15 pm +01:00">Mar 25, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -33,7 +33,9 @@ export default class ActivityPub {
     if (!activityPub) {
       dotenv.config()
       const url = new URL(process.env.GRAPHQL_URI)
-      activityPub = new ActivityPub(url.hostname || 'localhost', url.port || 4000, url.origin)
+      // TODO Check why the hostname attribute in the prod env not containing the tld! Following line is a quick fix!!
```


@mastercuber I would like to fix this *the clean way:tm:*

**Comment by Tirokk** [comment]
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-25T13:52:04Z" title="Monday, March 25th 2019, 2:52:04 pm +01:00">Mar 25, 2019</time>_

---


@Mastercuber I updated the description. You can close issues via keywords on Github.

File: backend/src/activitypub/ActivityPub.js (hunk 1 above)
Comment (Tirokk) [reply backend/src/activitypub/ActivityPub.js:36]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-03-25T13:56:22Z" title="Monday, March 25th 2019, 2:56:22 pm +01:00">Mar 25, 2019</time>_

---


https://nodejs.org/api/url.html#url_url_host this seems to be the clean fix.

**Review by Mogge — CHANGES_REQUESTED** [review]
@Mastercuber I think you need some help here. I'll jump in

**Review by Mogge — APPROVED** [review]

**Pull request closed without merging**